packages/core/src/2d/assembler/SlicedSpriteAssembler.ts

```diff
@@ -0,0 +1,147 @@
+import { Matrix, Vector2, Vector3 } from "@oasis-engine/math";
+import { SpriteMask } from "../sprite";
+import { SpriteRenderer } from "../sprite/SpriteRenderer";
+import { IAssembler } from "./IAssembler";
+import { StaticInterfaceImplement } from "./StaticInterfaceImplement";
+
+@StaticInterfaceImplement<IAssembler>()
+export class SlicedSpriteAssembler {
+  static _worldMatrix: Matrix = new Matrix();
+  static resetData(renderer: SpriteRenderer | SpriteMask): void {
+    const { _renderData: renderData } = renderer;
+    const { positions, uvs } = renderData;
+    if (positions.length < 16) {
+      for (let i = positions.length; i < 16; i++) {
+        positions.push(new Vector3());
+        uvs.push(new Vector2());
+      }
+    }
+    renderData.triangles = [];
+  }
+
+  static updateData(renderer: SpriteRenderer | SpriteMask): void {}
+
+  static updatePositions(renderer: SpriteRenderer | SpriteMask): void {
+    // Update WorldMatrix.
+    const { width, height, sprite } = renderer;
+    const { positions, uvs, triangles } = renderer._renderData;
+    const { border } = sprite;
+    const spriteUVs = sprite._getUVs();
+    // Update local positions.
+    const spritePositions = sprite._getPositions();
+    const { x: left, y: bottom } = spritePositions[0];
+    const { x: right, y: top } = spritePositions[3];
+    const expectWidth = sprite._getPixelWidth() / SpriteRenderer._pixelPerUnit;
+    const expectHeight = sprite._getPixelHeight() / SpriteRenderer._pixelPerUnit;
+    const fixedLeft = expectWidth * border.x;
```

Comment:
`expectWidth` and `expectHeight ` is opaque, developer can't prejudgment when is the Extrude boundary。

If Sprite have width and height(auto generate in constructor with texture), this is better for developer to know when is the Extrude boundary and can adjust sprite size